### DISCUSSION TITLE
How to customize validation message ?

### DISCUSSION THREAD

**Discussion by PubliAlex** [twbs/bootstrap#33259] [General]
Hello,

I don't see on the documentation how you can customize the error message depending on the user input.

Let's imagine this form, using bootstrap validation system (not default browser) : 

        
```
        <form class="needs-validation" novalidate>
            <div class="form-floating mb-3">
                <input class="form-control" type="email" id="field-email" name="email" placeholder="john.doe@provider.tld" required />
                <label class="form-label" for="field-email">Email</label>
                <div class="invalid-feedback">Email is required</div>
            </div>
            <button class="btn btn-primary" type="submit">Valid form</button>
        </form>
```

If I submit the form with email field empty : no problem, the returned message is relevant. But if I fill "test" in the email address, the field is not empty so the response "Email is required" is not correct, it should be something like "Email don't have the right format". I dont see on the documentation how can you set different messages depending on the error.

Thank you for your help

**Comment by florianlacreuse** [comment]
If I understand your problem correctly, this is not a job for Bootstrap. You will have to dynamically fill in the content inside the `<div class="invalid-feedback">` tag.
